vta/tests/python/tsim/tsim_alu.py

```diff
@@ -0,0 +1,168 @@
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+"""Test ALU operations"""
+
+######################################################################
+# RPC Setup
+# ---------
+# We start by programming the Pynq's FPGA and building its RPC runtime
+# as we did in the VTA introductory tutorial.
+
+from __future__ import absolute_import, print_function
+
+import argparse
+import sys
+import os
+import tvm
+import vta
+import numpy as np
+from tvm import rpc
+from tvm.contrib import util
+from vta.testing import simulator
+
+def check_alu(tid, tvm_op, np_op=None, use_imm=False):
+    """Test ALU"""
+
+    # Load VTA parameters from the vta/config/vta_config.json file
+    env = vta.get_env()
+
+    # We read the Pynq RPC host IP address and port number from the OS environment
+    host = os.environ.get("VTA_PYNQ_RPC_HOST", "192.168.2.99")
+    port = int(os.environ.get("VTA_PYNQ_RPC_PORT", "9091"))
+
+    # We configure both the bitstream and the runtime system on the Pynq
+    # to match the VTA configuration specified by the vta_config.json file.
+    if env.TARGET == "pynq":
+
+        # Make sure that TVM was compiled with RPC=1
+        assert tvm.module.enabled("rpc")
+        remote = rpc.connect(host, port)
+
+        # Reconfigure the JIT runtime
+        vta.reconfig_runtime(remote)
+
+        # Program the FPGA with a pre-compiled VTA bitstream.
+        # You can program the FPGA with your own custom bitstream
+        # by passing the path to the bitstream file instead of None.
+        vta.program_fpga(remote, bitstream=None)
+
+    # In simulation mode, host the RPC server locally.
+    elif env.TARGET == "sim" or env.TARGET == "tsim":
+        remote = rpc.LocalSession()
+
+    if env.TARGET == "tsim":
+        simulator.tsim_init("libvta_hw")
+
+    m = 8
+    n = 8
+    imm = np.random.randint(1,5)
+
+    # compute
+    a = tvm.placeholder(
+        (m, n, env.BATCH, env.BLOCK_OUT),
+        name="a",
+        dtype=env.acc_dtype)
+    a_buf = tvm.compute(
+        (m, n, env.BATCH, env.BLOCK_OUT),
+        lambda *i: a(*i),
+        "a_buf") #DRAM->SRAM
+    if use_imm:
+        res_buf = tvm.compute(
+            (m, n, env.BATCH, env.BLOCK_OUT),
+            lambda *i: tvm_op(a_buf(*i), imm),
+            "res_buf") #compute
+    else:
+        b = tvm.placeholder(
+            (m, n, env.BATCH, env.BLOCK_OUT),
+            name="b",
+            dtype=env.acc_dtype)
+        b_buf = tvm.compute(
+            (m, n, env.BATCH, env.BLOCK_OUT),
+            lambda *i: b(*i),
+            "b_buf") #DRAM->SRAM
+        res_buf = tvm.compute(
+            (m, n, env.BATCH, env.BLOCK_OUT),
+            lambda *i: tvm_op(a_buf(*i), b_buf(*i)),
+            "res_buf") #compute5B
```

Comment:
typo fix in comment?

Reply:
Same [comment](https://github.com/dmlc/tvm/pull/3258#discussion_r289252350)